Libraries/oneMKL/matrix_mul/matrix_mul_mkl.cpp

```diff
@@ -0,0 +1,184 @@
+//==============================================================
+// Copyright � 2020 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <limits>
+#include "mkl.h"
+#include "mkl_blas_sycl.hpp"
+
+using namespace std;
+using namespace cl::sycl;
+
+// Matrix size constants
+auto constexpr SIZE = 4800  // Must be a multiple of 8.
+auto constexpr M = SIZE / 8
+auto constexpr N = SIZE / 4
+auto constexpr P = SIZE / 2
+
+/**
+ * Perform the matrix multiplication on host to verify results from mkl.
+ */
+int VerifyResult(double *c_back);
+
+int main() {
+  //
+  // Initialize data for Gemm
+  //
+  // C = alpha * op(A) * op(B)  + beta * C
+  //
+  mkl::transpose transA = mkl::transpose::nontrans;
+  mkl::transpose transB = mkl::transpose::nontrans;
+
+  // matrix data sizes
+  int m = M;
```

Comment:
I wonder whether you need these 3 variables at all.
Or just remove `N` and just replace it by `n` directly. so `auto constexpr n = size/8;` etc.

Reply:
I keep thinking this is some useless obfuscation